Fix: Stackable Foods in Microwaves and Eating

Lennoxite opened the pull request:
# Stackable Food Fixing

### Purpose
- ~Eating a food with stackable now causes the stack to be eaten one at a time, instead of altogether. This modifies the player's eating function.~ Obsolete due to another PR.

- Microwaving stacked foods should now create a cooked item stack of equal amount. This modifies both the microwave component and the object.

- Removes EdibleStack, it is now obsolete. Cooked Cutlets now use Edible instead of EdibleStack.
### Notes:
- ~There is no public "increase stack" function in Stackable, so passing a negative value into ServerConsume was used instead when creating food stacks from the microwave.~ Added new function.

Lennoxite committed:
Merge pull request #2 from unitystation/develop

Another Merge

Lennoxite committed:
Modifies player eat function to eat stacked food one at a time. Makes microwave handle stacks. Renders EdibleStack obsolete.

Player eat function now eats stacked food one at a time, instead of the whole thing. (EdibleStack now obsolete, eliminated from code.)

Microwaves now make sure that stackable food amount in = stackable food amount out.

Lennoxite committed:
Added additional check to ensure that microwave doesn't delete the end stack entirely if mealCount == 0 (AKA starting item is not Stackable)

Lennoxite committed:
Give meal count an initial value to prevent initialization errors.

Lennoxite committed:
Revert PlayerNetworkActions.cs

PNA update no longer required due to Edible refactoring.

Lennoxite committed:
Merge branch 'develop' into foodstuffv2

fomalsd commented on UnityProject/Assets/Scripts/Machines/Microwave.cs:
```diff
@@ -104,9 +116,28 @@ private void FinishCooking()
+
+				if (stckChanger != 0)
+				{
+					stck.ServerConsume(-stckChanger);
```
I don't think ServerConsume is designed to work with negative values. Perhaps you could change `stckChanger != 0` to `stckChanger < 0` to make sure it's positive when inverted?

Lennoxite replied:
Is there another way to increase the stack amount of a newly stacked item? ServerConsume was implemented this way in order to ensure that a stackable item out has the same amount as the stackable item in (if applicable). The initial spawn amount for a cooked cutlet, for example, is 3. Both raw and cooked cutlets have a max size of 6. If I put in 4 raw cutlets, I would need to add 1 more to the cooked item amount, whereas if I put in 2, I'd have to subtract 1 from the product stack. ServerConsume can handle the second case, but I'd need another way to handle the first.

Right now, last time I checked, the only way to directly adjust the amount in a stackable item is through SyncAmount, a private function. Another option would be to spawn more stacks as needed of an object and use ServerCombine to add them all up into a single entity, and that seems like a bit more work than needed. Granted, I could just make a new ServerIncrease function for Stackable, but it might be best if I don't have to mess around with other files too much.

Lennoxite replied:
Decided to just make a ServerIncrease function for Stackable and had microwave implement it.

Lennoxite committed:
Adds Stackable.ServerIncrease function to increase amount outside the Stackable object, refactors Microwave to use new function.

Stackable.ServerIncrease does the inverse of ServerConsume, fill up by the given amount and do not go above maxAmount.

Lennoxite committed:
Merge branch 'foodstuffv2' of https://github.com/Lennoxite/unitystation into foodstuffv2
